ponyo/simulate_expression_data.py

```diff
@@ -199,12 +199,15 @@ def simulate_by_random_sampling(
 
 def simulate_by_latent_transformation(
     num_simulated_experiments,
-    normalized_data_file,
+    normalized_data_filename,
```

Comment:
One thing you could think about is changing these `simulate_by_X` functions to also take a DataFrame as input (i.e. `normalized_data` rather than `normalized_data_filename`). This would make them consistent with `shift_template_experiment`, as well as possibly making them faster (although I'm not sure if you're calling these ones in a loop anywhere).

Might be a non-trivial amount of work to update other projects to conform with this, though, so if you want to create a separate issue and think about it later that's fine.

Reply:
Good catch! I was waffling about this too. So the other two methods don't run in a for loop so its not required that we use the dataframe. I left them because it makes more sense me to input filenames instead of dataframes (i know its an extra step but feels like its one less hassle) and this one method is more of an exception.

I'll add an issue. One thing I could see doing is making a wrapper function to call the for loop